### ISSUE TITLE
[Vivaldi] Unable to fetch bookmarks from Nextcloud 12 instance

### ISSUE THREAD

**Issue by e-alfred** [marcelklehr/floccus#18]
Hello,

I installed the Floccus 1.2.0 *.crx plugin in Vivaldi 1.10.867.38 and entered my address/credentials, but the plugin is unable to fetch Bookmarks from a Nextcloud 12 instance with the Bookmarks app 0.10.0 enabled.

I get the following in the developer console for the plugin in Vivaldi:

```
Fetching bookmarks Objectpassword: "password"url: "https://server/owncloud/"username: "user"__proto__: Object
server/owncloud//index.php/apps/bookmarks/public/rest/v2/bookmark?page=-1 Failed to load resource: the server responded with a status of 500 (Internal Server Error)
background-script.js:100 Error: Failed to retrieve bookmarks from ownCloud
    at browser.storage.local.get.then.then.response (background-script.js:23)
    at <anonymous>
bookmarks.sync.catch.err @ background-script.js:100
```

This is the log entry in the nextcloud.log file on the server:

```
{"reqId":"grhrhf","level":3,"time":"2017-06-21T15:33:35+02:00","remoteAddr":"1.1.1.1","user":"--","app":"index","method":"GET","url":"\/owncloud\/\/index.php\/apps\/bookmarks\/public\/rest\/v2\/bookmark","message":"Exception: {\"Exception\":\"Exception\",\"Message\":\"The requested uri(\\\/owncloud\\\/index.php\\\/apps\\\/bookmarks\\\/public\\\/rest\\\/v2\\\/bookmark) cannot be processed by the script '\\\/owncloud\\\/\\\/index.php')\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/lib\\\/base.php(952): OC\\\\AppFramework\\\\Http\\\\Request->getRawPathInfo()\\n#1 \\\/var\\\/www\\\/owncloud\\\/index.php(40): OC::handleRequest()\\n#2 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Request.php\",\"Line\":729}","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/59.0.3071.90 Safari\/537.36 Vivaldi\/1.91.867.38","version":"12.0.0.29"}
{"reqId":"grhrhf","level":0,"time":"2017-06-21T15:35:15+02:00","remoteAddr":"1.1.1.1","user":"--","app":"core","method":"GET","url":"\/owncloud\/\/index.php\/apps\/bookmarks\/public\/rest\/v2\/bookmark?page=-1","message":"Scss is disabled for \/var\/www\/owncloud\/core\/css\/styles.scss, ignoring","userAgent":"Mozilla\/5.0 (X11; Linux x86_64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/59.0.3071.90 Safari\/537.36 Vivaldi\/1.91.867.38","version":"12.0.0.29"}
```

**Comment by marcelklehr** [comment]
Could you try removing the trailing slash from the root URL?
--
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by e-alfred** [comment]
Okay, removed the trailing slash and now it fetches the bookmarks from the server as I can see in the console. But no bookmarks are shown in the Bookmarks pane of Vivaldi. Not even the Owncloud folder is created by the extension. I cannot see any error in the console though.

**Comment by marcelklehr** [comment]
The owncloud folder should be somewhere. It's a pattern that people don't look to hard for it. It might not be toplevel. In firefox it's inside "Other bookmarks"

**Comment by e-alfred** [comment]
I tried it on two different Vivaldi setups on different machines, but neither the folder gets created nor any bookmarks get synced. If I search for them on a fresh Vivaldi installation, none of them are found by the search nor are they showing up anywhere. 

It seems that Vivaldi does something different compared to other Chromium-based browsers, soI sent a mail to their support as suggested here:

https://help.vivaldi.com/article/extensions/

I will tell you if I get some more information from them.

**Comment by e-alfred** [comment]
I got this answer back from Vivaldi support:


> There are extensions that still doesn't work well with Vivaldi especially those that interacts with bookmarks, UI and history since Vivaldi does not use the same format as Chrome does. I will look into this extension, but it would be helpful if you contact the developer and ask for Vivaldi support. We will do the same.

**Comment by lala-rob** [comment]
I found something interesting when i export my Bookmarks.
The Sync is working but the Folder is not visible.
Vivaldi is storing his Bookmarks in an Subfolder(Bookmarks).

Maby you must only change the root folder...
![2017-08-21 09_19_43-lesezeichen](https://user-images.githubusercontent.com/2798187/29508035-35da9d3a-8653-11e7-990b-a7b6cf2ad758.png)

**Comment by evook** [comment]
I  can confirm @lala-rob findings.

**Comment by marcelklehr** [comment]
I will probably have to add the ability to select a custom folder then.

**Comment by e-alfred** [comment]
Tried 1.3.0 beta 3 but the problem persists. The bookmarks get downloaded without a problem but no folder is created nor do the bookmarks show up.

**Comment by marcelklehr** [comment]
Let me know if this is fixed in floccus v2.0.0. You should be able to select a custom folder now ;)

**Comment by e-alfred** [comment]
Yes, selection of a different folder resolves the bug! Thanks for the awesome work!
